src/app/chat/chat.service.ts

```diff
@@ -128,29 +91,74 @@ export class ChatService {
    *  size:20
    * }
    */
-  getMessageList(prams: messageListPrams): Observable<any> {
-    console.log("Service", this.messageList);
-    return of(this.messageList);
+  getMessageList(data: messageListPrams, selectedChat: any): Observable<any> {
+    return this.request
+      .get(api.getChatMessages, {
+        params: data
+      })
+      .pipe(
+        map(response => {
+          if (response.success && response.data) {
+            return this._normalisMessageListResponse(
+              response.data,
+              selectedChat
+            );
+          }
+        })
+      );
   }
 
   markMessagesAsSeen(prams: markAsSeenPrams): Observable<any> {
-    prams.action = 'mark_seen';
-    return of("maked");
+    let body = {
+      team_id: prams.team_id,
+      id: prams.id,
+      action: "mark_seen"
+    };
+    return this.request.post(api.markAsSeen, body, {
+      headers: { "Content-Type": "application/x-www-form-urlencoded" }
+    });
   }
 
   postNewMessage(data: newMessage): Observable<any> {
-    data.env = 'develop';
-    let returnData = {
-      id: 300,
-      sender_name: 'Chathumal',
-      is_sender: true,
-      team_name: 'dream team',
-      sent_time: '1.30 PM',
-      message: data.message,
-      team_id: data.team_id,
-      to: data.to
+    const body = new HttpParams()
+      .set("data[to]", data.to.toString())
+      .set("data[message]", data.message)
+      .set("data[team_id]", data.team_id.toString())
+      .set("env", environment.env);
+      if (data.participants_only) {
+        body.set("participants_only", data.participants_only.toString());
+      }
+    return this.request.post(api.createMessage, body.toString(), {
+      headers: { "Content-Type": "application/x-www-form-urlencoded" }
+    });
+  }
+
+  unreadMessageCout(data: unreadMessagePrams): Observable<any> {
+    let body = {
+      unread_count_for: data.filter
     };
-    return of(returnData);
+    return this.request.get(api.getChatMessages, body);
+  }
+
+  getTeamName(id: number): Observable<any> {
+    let data = {
+      team_id: id
+    };
+    return this.request
+      .get(api.getTeam, {
+        params: data
+      })
+      .pipe(
+        map(response => {
+          if (response.success && response.data) {
+            return this._normalisTeamResponse(response.data);
+          }
+        })
+      );
+  }
+
+  private _normalisTeamResponse(response) {
+    return response.Team.name;
   }
 
   generateChatAvatarText(text) {
```

Comment:
As avatar text will show first two letters of `chatNameArray[1]`.
if `chatNameArray[0]` have value and `chatNameArray[1]` null. it will get first two letters of `chatNameArray[0]`.